File: cases/ProjetoTestC/src/handler.c
/*
 * Trata a requisição já roteada e aciona a execução do comando.
 *
 * Elo do caminho vulnerável: handler_execute -> run_command.
 */

#include "../include/exec.h"
#include "../include/handler.h"
#include "../include/logger.h"

/* Extrai o conteúdo da requisição, removendo a quebra de linha final.
 *
 * Não participa do caminho vulnerável: apenas prepara o texto. */
static char *parse_payload(char *request)
{
    char *cursor = request;

    while (*cursor != '\0') {
        if (*cursor == '\n') {
            *cursor = '\0';
            break;
        }

        cursor++;
    }

    return request;
}

void handler_execute(const char *request)
{
    char *payload = parse_payload((char *)request);

    if (payload_is_empty(payload)) {
        log_error("Requisicao sem conteudo.");
        return;
    }

    run_command(payload);
}

File: cases/ProjetoTestPython/src/handler.py
"""Trata a requisição já roteada e aciona a execução do comando.

Elo do caminho vulnerável: handler_execute -> run_command.
"""

from executor import run_command
from logger import log_error


def parse_payload(request):
    """Remove espaços e a quebra de linha final da requisição.

    Não participa do caminho vulnerável: apenas prepara o texto.
    """
    return request.strip()


def payload_is_empty(payload):
    """Informa se o conteúdo recebido está vazio."""
    return payload is None or payload == ""


def handler_execute(request):
    """Prepara o conteúdo e aciona a execução."""
    payload = parse_payload(request)

    if payload_is_empty(payload):
        log_error("Requisicao sem conteudo.")
        return None

    return run_command(payload)
